chore(samples): add samples for for Retail Search service

t-karasova opened the pull request:
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-retail/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

t-karasova committed:
feat: Code samples for Retail Tutorials. Search service samples

t-karasova committed:
feat: Code samples for Retail Tutorials. Search service samples

t-karasova committed:
Revert "feat: Code samples for Retail Tutorials. Search service samples"

This reverts commit 153c92d2

kurtisvg commented on .kokoro/common.cfg:
```diff
@@ -11,3 +11,12 @@ env_vars: {
   key: "TRAMPOLINE_BUILD_FILE"
   value: "github/java-retail/.kokoro/build.sh"
 }
+env_vars: {
+  key: "BUCKET_NAME"
+  value: "crs-interactive-tutorials"
+}
+
+env_vars: {
+  key: "EVENTS_BUCKET_NAME"
+  value: "crs-interactive-tutorials"
+}
```
These kokoro files are templated, so you probably want to rely on this file to populate secrets: https://github.com/googleapis/java-retail/blob/main/.kokoro/populate-secrets.sh

t-karasova replied:
done

kurtisvg commented on .kokoro/continuous/common.cfg:
```diff
@@ -20,6 +20,16 @@ env_vars: {
 env_vars: {
-  key: "JOB_TYPE"
-  value: "test"
+  key: "PROJECT_NUMBER"
+  value: "945579214386"
+}
+
+env_vars: {
+  key: "BUCKET_NAME"
+  value: "crs-interactive-tutorials"
+}
+
+env_vars: {
+  key: "EVENTS_BUCKET_NAME"
+  value: "crs-interactive-tutorials"
```
Same with this file (see https://github.com/googleapis/java-retail/blob/main/.kokoro/populate-secrets.sh)

t-karasova replied:
Hi, Kurtis, @kurtisvg, kould you please help me to understand, if I need to populate the secrets from the secrets manager  from service account: kokoro-trampoline@cloud-devrel-kokoro-resources.iam.gserviceaccount.com, how can I add our secrets there? Should I request assistance from some team?

kurtisvg replied:
@chingor13 knows more about how to add secrets, although the correct process might just be to ask someone with permissions.

@lesv FYI

t-karasova replied:
done

kurtisvg commented on samples/snippets/README.md:
````diff
@@ -0,0 +1,29 @@
+In order to authenticate and authorize the client place a JSON file with authorization token to your cloudshell project and refer environment variable `GOOGLE_APPLICATION_CREDENTIALS` to it (use full path to file):
+
+```
+export GOOGLE_APPLICATION_CREDENTIALS="/home/use_name/cloudshell_open/grs-retail/token.json"
````
Generally we encourage folks to just use [Application Default Credentials (ADC)](https://cloud.google.com/docs/authentication/best-practices-applications)

t-karasova replied:
It is recommended for Retail clients to authenticate using GOOGLE_APPLICATION_CREDENTIALS

kurtisvg commented on samples/snippets/README.md:
````diff
@@ -0,0 +1,29 @@
+```
+
+### Define environment variable
+Set PROJECT_NUMBER value as an environment variable.
````
There are some other variables specified in your kokoro configs. Are they required?

t-karasova replied:
they will be used for Product and UserEvents services

kurtisvg commented on samples/snippets/pom.xml:
```diff
@@ -1,47 +1,37 @@
-<?xml version='1.0' encoding='UTF-8'?>
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<?xml version="1.0" encoding="UTF-8"?>
```
This file is generated too, so you may need to opt it out - @chingor13 for guidance here

t-karasova replied:
the original code restored and all other project settings are added

kurtisvg commented on samples/snippets/pom.xml:
```diff
@@ -1,47 +1,37 @@
     <dependency>
       <groupId>junit</groupId>
       <artifactId>junit</artifactId>
       <version>4.13.2</version>
```
Why isn't this a test scope?

kurtisvg commented on samples/snippets/pom.xml:
```diff
@@ -1,47 +1,37 @@
       <groupId>com.google.cloud</groupId>
       <artifactId>google-cloud-retail</artifactId>
-      <version>2.0.6</version>
+      <version>2.0.3</version>
```
Why are we downgrading the version?

t-karasova replied:
fixed

kurtisvg commented on samples/snippets/pom.xml:
```diff
@@ -1,47 +1,37 @@
-      <artifactId>truth</artifactId>
-      <version>1.1.3</version>
-      <scope>test</scope>
+      <groupId>io.grpc</groupId>
```
Why is GRPC a dependency?

t-karasova replied:
removed

kurtisvg commented on samples/snippets/src/main/java/search/SearchSimpleQuery.java:
```diff
@@ -0,0 +1,81 @@
+import java.io.IOException;
+import java.util.UUID;
+
+public class SearchSimpleQuery {
```
Please make sure these samples follow the [sample style guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)

t-karasova replied:
checked

kurtisvg reviewed (changes requested):
There are quite a few samples here (currently +1k lines) - please consider breaking them into smaller PRs to make them easier to review.

t-karasova committed:
code style fix, the retail-samples-secrets is added as SECRET_MANAGER_KEYS

t-karasova committed:
code style fix

renovate-bot committed:
build(deps): update dependency com.google.cloud:google-cloud-shared-config to v1.2.4 (#257)

[![WhiteSource Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [com.google.cloud:google-cloud-shared-config](https://togithub.com/googleapis/java-shared-config) | `1.2.2` -> `1.2.4` | [![age](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-config/1.2.4/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-config/1.2.4/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-config/1.2.4/compatibility-slim/1.2.2)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-config/1.2.4/confidence-slim/1.2.2)](https://docs.renovatebot.com/merge-confidence/) |

---

### Release Notes

<details>
<summary>googleapis/java-shared-config</summary>

### [`v1.2.4`](https://togithub.com/googleapis/java-shared-config/blob/HEAD/CHANGELOG.md#&#8203;124-httpswwwgithubcomgoogleapisjava-shared-configcomparev123v124-2022-01-06)

[Compare Source](https://togithub.com/googleapis/java-shared-config/compare/v1.2.3...v1.2.4)

### [`v1.2.3`](https://togithub.com/googleapis/java-shared-config/blob/HEAD/CHANGELOG.md#&#8203;123-httpswwwgithubcomgoogleapisjava-shared-configcomparev122v123-2022-01-04)

[Compare Source](https://togithub.com/googleapis/java-shared-config/compare/v1.2.2...v1.2.3)

</details>

---

### Configuration

📅 **Schedule**: At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update again.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, click this checkbox.

---

This PR has been generated by [WhiteSource Renovate](https://renovate.whitesourcesoftware.com). View repository job log [here](https://app.renovatebot.com/dashboard#github/googleapis/java-retail).

chenlei1216 commented on samples/snippets/src/main/java/search/SearchSimpleQuery.java:
```diff
@@ -0,0 +1,130 @@
+  private static SearchServiceClient getSearchServiceClient()
+      throws IOException {
+    SearchServiceSettings settings = SearchServiceSettings.newBuilder()
+        .setEndpoint(ENDPOINT)
```
This endpoint is the default value so it can be ignored:

return SearchServiceClient.create();

t-karasova replied:
fixed

chenlei1216 commented on samples/snippets/src/main/java/search/SearchWithBoostSpec.java:
```diff
@@ -0,0 +1,144 @@
+   */
+  public static SearchResponse search() throws IOException {
+    // TRY DIFFERENT CONDITIONS HERE:
+    String condition = "(colorFamily: ANY(\"Blue\"))";
```
colorFamilies

chenlei1216 commented on samples/snippets/src/main/java/search/SearchWithFiltering.java:
```diff
@@ -0,0 +1,142 @@
+   */
+  public static SearchResponse search() throws IOException {
+    // TRY DIFFERENT FILTER EXPRESSIONS HERE:
+    String filter = "(colorFamily: ANY(\"Black\"))";
```
colorFamilies

t-karasova replied:
done

renovate-bot committed:
deps: update dependency com.google.cloud:google-cloud-shared-dependencies to v2.6.0 (#258)

[![WhiteSource Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [com.google.cloud:google-cloud-shared-dependencies](https://togithub.com/googleapis/java-shared-dependencies) | `2.5.1` -> `2.6.0` | [![age](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-dependencies/2.6.0/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-dependencies/2.6.0/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-dependencies/2.6.0/compatibility-slim/2.5.1)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-shared-dependencies/2.6.0/confidence-slim/2.5.1)](https://docs.renovatebot.com/merge-confidence/) |

---

### Release Notes

<details>
<summary>googleapis/java-shared-dependencies</summary>

### [`v2.6.0`](https://togithub.com/googleapis/java-shared-dependencies/blob/HEAD/CHANGELOG.md#&#8203;260-httpswwwgithubcomgoogleapisjava-shared-dependenciescompare251v260-2022-01-07)

[Compare Source](https://togithub.com/googleapis/java-shared-dependencies/compare/v2.5.1...v2.6.0)

##### Bug Fixes

-   **java:** add -ntp flag to native image testing command ([#&#8203;1299](https://www.togithub.com/googleapis/java-shared-dependencies/issues/1299)) ([#&#8203;546](https://www.togithub.com/googleapis/java-shared-dependencies/issues/546)) ([d8bb8d9](https://www.github.com/googleapis/java-shared-dependencies/commit/d8bb8d9d8e3b8c582f768cd3164e8580ec334511))
-   **java:** run Maven in plain console-friendly mode ([#&#8203;1301](https://www.togithub.com/googleapis/java-shared-dependencies/issues/1301)) ([#&#8203;557](https://www.togithub.com/googleapis/java-shared-dependencies/issues/557)) ([6b57d08](https://www.github.com/googleapis/java-shared-dependencies/commit/6b57d08816ec6c1dfb637cae022624e896930ad8))

##### Dependencies

-   gRPC upgrade to 1.43.2 ([#&#8203;567](https://www.togithub.com/googleapis/java-shared-dependencies/issues/567)) ([c946655](https://www.github.com/googleapis/java-shared-dependencies/commit/c94665571f26d24fbbb3f89ef5a3db99663bfdba))
-   update dependency com.google.api-client:google-api-client-bom to v1.33.0 ([#&#8203;563](https://www.togithub.com/googleapis/java-shared-dependencies/issues/563)) ([0aedddb](https://www.github.com/googleapis/java-shared-dependencies/commit/0aedddb9a535753767d56f5910f355c7a1c8509c))
-   update dependency com.google.api:api-common to v2.1.2 ([2f154a3](https://www.github.com/googleapis/java-shared-dependencies/commit/2f154a3ff759211438c72859e98b314c7ff8405f))
-   update dependency com.google.http-client:google-http-client-bom to v1.41.0 ([#&#8203;562](https://www.togithub.com/googleapis/java-shared-dependencies/issues/562)) ([6da0735](https://www.github.com/googleapis/java-shared-dependencies/commit/6da0735d4db9df998d11c549488a024b800bed65))
-   update dependency io.grpc:grpc-bom to v1.43.1 ([#&#8203;553](https://www.togithub.com/googleapis/java-shared-dependencies/issues/553)) ([b787c8f](https://www.github.com/googleapis/java-shared-dependencies/commit/b787c8f2032b7ffc9797158c3ff214b7813f34b7))
-   update dependency org.apache.httpcomponents:httpcore to v4.4.15 ([#&#8203;550](https://www.togithub.com/googleapis/java-shared-dependencies/issues/550)) ([e2d3dba](https://www.github.com/googleapis/java-shared-dependencies/commit/e2d3dbad6dd54c117abb5ab1b6137cfee4e5a32b))
-   update dependency org.checkerframework:checker-qual to v3.20.0 ([#&#8203;549](https://www.togithub.com/googleapis/java-shared-dependencies/issues/549)) ([586df59](https://www.github.com/googleapis/java-shared-dependencies/commit/586df5949d4ddca11d0705d696882c32d8436c31))
-   update gax.version to v2.8.0 ([#&#8203;564](https://www.togithub.com/googleapis/java-shared-dependencies/issues/564)) ([9ad5d4f](https://www.github.com/googleapis/java-shared-dependencies/commit/9ad5d4f5385753a0c5739ef400f5d346e3342b84))
-   update gax.version to v2.8.1 ([2f154a3](https://www.github.com/googleapis/java-shared-dependencies/commit/2f154a3ff759211438c72859e98b314c7ff8405f))
-   update google.common-protos.version to v2.7.1 ([#&#8203;561](https://www.togithub.com/googleapis/java-shared-dependencies/issues/561)) ([9aa860d](https://www.github.com/googleapis/java-shared-dependencies/commit/9aa860d92f745f1d4382eb16439bf17ac6ca8ae9))
-   update google.core.version to v2.3.5 ([#&#8203;565](https://www.togithub.com/googleapis/java-shared-dependencies/issues/565)) ([2f154a3](https://www.github.com/googleapis/java-shared-dependencies/commit/2f154a3ff759211438c72859e98b314c7ff8405f))
-   update iam.version to v1.2.0 ([#&#8203;547](https://www.togithub.com/googleapis/java-shared-dependencies/issues/547)) ([3232c82](https://www.github.com/googleapis/java-shared-dependencies/commit/3232c828b628cb604a200b3e569e56e758e5e11d))
-   upgrading protobuf to 3.19.2 ([#&#8203;560](https://www.togithub.com/googleapis/java-shared-dependencies/issues/560)) ([7ef3325](https://www.github.com/googleapis/java-shared-dependencies/commit/7ef33250f5473348d3546210b7a534360f476101))

##### [2.5.1](https://www.github.com/googleapis/java-shared-dependencies/compare/2.5.0...v2.5.1) (2021-12-03)

##### Dependencies

-   update dependency com.google.api:api-common to v2.1.1 ([#&#8203;533](https://www.togithub.com/googleapis/java-shared-dependencies/issues/533)) ([4e23adb](https://www.github.com/googleapis/java-shared-dependencies/commit/4e23adb666dfaa18ae8bf869aca030153de16d62))
-   update gax.version to v2.7.1 ([#&#8203;544](https://www.togithub.com/googleapis/java-shared-dependencies/issues/544)) ([5a0c480](https://www.github.com/googleapis/java-shared-dependencies/commit/5a0c480671fa834e600edc6ac6fe77967d1b79b3))
-   update google.common-protos.version to v2.7.0 ([#&#8203;542](https://www.togithub.com/googleapis/java-shared-dependencies/issues/542)) ([af97eb5](https://www.github.com/googleapis/java-shared-dependencies/commit/af97eb5d9cff397080fdb3d4a6dbf7c2692e0358))
-   update google.core.version to v2.3.3 ([#&#8203;534](https://www.togithub.com/googleapis/java-shared-dependencies/issues/534)) ([885acef](https://www.github.com/googleapis/java-shared-dependencies/commit/885acefe4ff5313eb53dc43b4825ca597b97dcfd))

</details>

---

### Configuration

📅 **Schedule**: At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update again.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, click this checkbox.

---

This PR has been generated by [WhiteSource Renovate](https://renovate.whitesourcesoftware.com). View repository job log [here](https://app.renovatebot.com/dashboard#github/googleapis/java-retail).

release-please[bot] committed:
chore: release 2.0.7 (#253)

:robot: I have created a release \*beep\* \*boop\*
---
### [2.0.7](https://www.github.com/googleapis/java-retail/compare/v2.0.6...v2.0.7) (2022-01-07)


### Bug Fixes

* **java:** add -ntp flag to native image testing command ([#1299](https://www.github.com/googleapis/java-retail/issues/1299)) ([#250](https://www.github.com/googleapis/java-retail/issues/250)) ([dd942ab](https://www.github.com/googleapis/java-retail/commit/dd942ab9d8efd582f5c83c259c5ca6e0225c88a5))


### Dependencies

* update dependency com.google.cloud:google-cloud-shared-dependencies to v2.6.0 ([#258](https://www.github.com/googleapis/java-retail/issues/258)) ([81f1a09](https://www.github.com/googleapis/java-retail/commit/81f1a0936e779e67d39aa4d68f19b99bea90a819))
---


This PR was generated with [Release Please](https://github.com/googleapis/release-please). See [documentation](https://github.com/googleapis/release-please#release-please).

release-please[bot] committed:
chore: release 2.0.8-SNAPSHOT (#260)

:robot: I have created a release \*beep\* \*boop\*
---
### Updating meta-information for bleeding-edge SNAPSHOT release.
---


This PR was generated with [Release Please](https://github.com/googleapis/release-please). See [documentation](https://github.com/googleapis/release-please#release-please).

t-karasova committed:
Add search samples impl

Minor fixes

added product and events services

fixes according to the review comments

t-karasova committed:
Add search samples impl

added product and events services

Refactoring

Checkstyle-related refactoring.

Changes for fixing code review comments.

Conflicts are resolved

Add search samples impl

added product and events services

Refactoring

Checkstyle-related refactoring.

Changes for fixing code review comments.

Changes by code review comments.

Update ImportProductsBigQueryTable.java

t-karasova committed:
Add search samples impl

added product and events services

Refactoring

Checkstyle-related refactoring.

Changes for fixing code review comments.

t-karasova committed:
Add search samples impl

added product and events services

Refactoring

Checkstyle-related refactoring.

Changes for fixing code review comments.

t-karasova committed:
Conflicts are resolved

Add search samples impl

added product and events services

Refactoring

Checkstyle-related refactoring.

Changes for fixing code review comments.

chenlei1216 commented on samples/snippets/src/main/java/product/ImportProductsBigQueryTable.java:
```diff
@@ -0,0 +1,186 @@
+        getImportProductsBigQueryRequest(reconciliationMode);
+
+    OperationFuture<ImportProductsResponse, ImportMetadata> bigQueryOperation =
+        getProductServiceClient().importProductsAsync(importBigQueryRequest);
```
One caveat for using operationFuture is that this is returning a Java future which will expire in 5 minutes, after which future.get() will return a CancelledException, and product import usually will normally take more than 5 minutes. We can work around this by pulling the Operation manually. Example for this new style:

        ProductServiceClient serviceClient = getProductServiceClient();
        String operationName = serviceClient
                .importProductsCallable()
                .call(importBigQueryRequest)
                .getName();
        System.out.printf("OperationName = %s", operationName);
        OperationsClient operationsClient = serviceClient.getOperationsClient();
        Operation operation = operationsClient.getOperation(operationName);
        while (!operation.getDone()) {
            // Keep polling the operation periodically until the import task is done.
            Thread.sleep(60000);
            operation = operationsClient.getOperation(operationName);
        }
        if (operation.hasMetadata()) {
            ImportMetadata metadata = operation.getMetadata().unpack(ImportMetadata.class);
            System.out.printf("Number of successfully imported products: %s\n",
                    metadata.getSuccessCount());
            System.out.printf("Number of failures during the importing: %s\n",
                    metadata.getFailureCount());
        }
        if (operation.hasResponse()) {
            ImportProductsResponse response = operation.getResponse().unpack(ImportProductsResponse.class);
            System.out.printf("Number of successfully imported products: %s\n", response);
        }

t-karasova replied:
done

chenlei1216 commented on samples/snippets/src/main/java/product/RemoveFulfillmentPlaces.java:
```diff
@@ -0,0 +1,145 @@
+  /**
+   * This variable describes defined product id for field setting.
+   */
+  private static final String PRODUCT_ID = "remove_fulfillment_test_product_id";
```
Use an UUID or use tryToDeleteProductIfExists before the test?

t-karasova replied:
it was decided to use the same approach for SetInventory, AddFulfillmentInfo and RemoveFulfillmentInfo code samples, which is using tryToDeleteProductIfExists

chenlei1216 commented on samples/snippets/src/main/java/product/SetInventory.java:
```diff
@@ -0,0 +1,181 @@
+  /**
+   * This variable describes defined product id for field setting.
+   */
+  private static final String PRODUCT_ID = "inventory_test_product_id";
```
Use an UUID? May not be necessary since you use tryToDeleteProductIfExists.

chenlei1216 commented on samples/snippets/src/main/java/events/ImportUserEventsBigQuery.java:
```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2022 Google Inc.
```
nit: Google LLC now since 2017 :)

t-karasova replied:
fixed :)

t-karasova commented on samples/snippets/src/main/java/product/AddFulfillmentPlaces.java:
```diff
@@ -0,0 +1,142 @@
+  /**
+   * This variable describes defined product id for field setting.
+   */
+  private static final String PRODUCT_ID = "add_fulfillment_test_product_id";
```
the tryToDeleteProductIfExists method was added

t-karasova commented on samples/snippets/src/main/java/product/CrudProduct.java:
```diff
@@ -0,0 +1,239 @@
+  /**
+   * This variable describes defined product id for field setting.
+   */
+  private static final String PRODUCT_ID = "crud_product_id";
```
the tryToDeleteProductIfExists method was added

t-karasova commented:
@chenlei1216 Hi Lei, could you please check the conversations, maybe some of them could be resolved?

chenlei1216 commented on samples/snippets/src/main/java/events/ImportUserEventsBigQuery.java:
```diff
@@ -0,0 +1,189 @@
+      // Polling operation delay until the import task is done.
+      final int awaitDuration = 30;
+
+      getUserEventsServiceClient().awaitTermination(awaitDuration,
```
May just do serviceClient.awaitTermination.

Not sure if awaitTermination is the right approach though for tracking long running operations. Probably Thread.sleep is easier.

t-karasova replied:
done

chenlei1216 commented on samples/snippets/src/main/java/events/ImportUserEventsInline.java:
```diff
@@ -0,0 +1,198 @@
+    UserEventServiceClient serviceClient = getUserEventsServiceClient();
+
+    String operationName = serviceClient
+        .importUserEventsCallable()
```
Inline import for user events/products are generally fast, we may use the old style of using
serviceClient.importUserEventsAsync(importInlineRequest) to return the OperationFuture.

t-karasova replied:
done

chenlei1216 commented on samples/snippets/src/main/java/product/AddFulfillmentPlaces.java:
```diff
@@ -135,8 +150,8 @@ public static void main(final String[] args)
     getProduct(PRODUCT_NAME);
 
-    deleteProduct(PRODUCT_NAME);
+    // deleteProduct(PRODUCT_NAME);
```
We may still delete it.

t-karasova replied:
done

chenlei1216 commented on samples/snippets/src/main/java/product/RemoveFulfillmentPlaces.java:
```diff
@@ -137,8 +139,6 @@ public static void main(final String[] args)
 
     getProduct(PRODUCT_NAME);
-
-    deleteProduct(PRODUCT_NAME);
```
We can still delete the product

t-karasova replied:
done

chenlei1216 commented on samples/snippets/src/main/java/product/SetInventory.java:
```diff
@@ -173,9 +172,7 @@ public static void main(final String[] args)
 
     getProduct(PRODUCT_NAME);
-
-    deleteProduct(PRODUCT_NAME);
```
We can still delete the product

t-karasova replied:
done

sborisenkox committed:
Changes by code review comments.

t-karasova committed:
Update ImportProductsBigQueryTable.java

sborisenkox committed:
Fix modules dependencies

sborisenkox committed:
Minor changes.

sborisenkox committed:
Add unit tests for setup packages.

t-karasova committed:
Merge branch 'tutorials' into tutorials-fixes

t-karasova committed:
Merge pull request #3 from t-karasova/tutorials-fixes

Tutorials fixes

t-karasova commented:
I close this pull request as it becomes too big and not suitable for review. The first part of the code samples for Retail Tutorials can be found here https://github.com/googleapis/java-retail/pull/271
